### ISSUE TITLE
error on windows

### ISSUE THREAD

**Issue by carlodri** [conda-forge/protobuf-feedstock#133]
- [x] I read [the conda-forge documentation](https://conda-forge.org/docs/user/introduction.html#how-can-i-install-packages-from-conda-forge) and could not find the solution for my problem there.

Issue:

Working on Windows, see info below, I get the following:

```bash
$ python -c "from google.protobuf.empty_pb2 import Empty; Empty.FromString(b'')"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\Users\laboratorio\Miniconda3\envs\protobuf_env_mamba\lib\site-packages\google\protobuf\empty_pb2.py", line 5, in <module>
    from google.protobuf import descriptor as _descriptor
  File "C:\Users\laboratorio\Miniconda3\envs\protobuf_env_mamba\lib\site-packages\google\protobuf\descriptor.py", line 47, in <module>
    from google.protobuf.pyext import _message
ImportError: cannot import name '_message' from 'google.protobuf.pyext' (C:\Users\laboratorio\Miniconda3\envs\protobuf_env_mamba\lib\site-packages\google\protobuf\pyext\__init__.py)
```

This is just a minimum code to reproduce the problem. The same error comes out when I try to run the grpc helloworld example (see [here](https://www.grpc.io/docs/languages/python/quickstart/))

Note that I have set the env variable `PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=cpp`

PR  #132 could be a solution? I'm going to reopen it for testing.

<br/>
Environment (<code>mamba list</code>):
<details>

```
$ mamba list
# Name                    Version                   Build  Channel
bzip2                     1.0.8                h8ffe710_4    conda-forge
ca-certificates           2021.10.8            h5b45459_0    conda-forge
libffi                    3.4.2                h8ffe710_5    conda-forge
libprotobuf               3.19.1               h7755175_0    conda-forge
libzlib                   1.2.11            h8ffe710_1013    conda-forge
openssl                   3.0.0                h8ffe710_2    conda-forge
pip                       21.3.1             pyhd8ed1ab_0    conda-forge
protobuf                  3.19.1          py310h8a704f9_1    conda-forge
python                    3.10.0          hcf16a7b_3_cpython    conda-forge
python_abi                3.10                    2_cp310    conda-forge
setuptools                59.4.0          py310h5588dad_0    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
sqlite                    3.37.0               h8ffe710_0    conda-forge
tk                        8.6.11               h8ffe710_1    conda-forge
tzdata                    2021e                he74cb21_0    conda-forge
ucrt                      10.0.20348.0         h57928b3_0    conda-forge
vc                        14.2                 hb210afc_5    conda-forge
vs2015_runtime            14.29.30037          h902a5da_5    conda-forge
wheel                     0.37.0             pyhd8ed1ab_1    conda-forge
xz                        5.2.5                h62dcd97_1    conda-forge
zlib                      1.2.11            h8ffe710_1013    conda-forge
```
</details>

<br/>
Details about  <code>mamba</code> and system ( <code>mamba info</code> ):
<details>

```
$ mamba info

                  __    __    __    __
                 /  \  /  \  /  \  /  \
                /    \/    \/    \/    \
███████████████/  /██/  /██/  /██/  /████████████████████████
              /  / \   / \   / \   / \  \____
             /  /   \_/   \_/   \_/   \    o \__,
            / _/                       \_____/  `
            |/
        ███╗   ███╗ █████╗ ███╗   ███╗██████╗  █████╗
        ████╗ ████║██╔══██╗████╗ ████║██╔══██╗██╔══██╗
        ██╔████╔██║███████║██╔████╔██║██████╔╝███████║
        ██║╚██╔╝██║██╔══██║██║╚██╔╝██║██╔══██╗██╔══██║
        ██║ ╚═╝ ██║██║  ██║██║ ╚═╝ ██║██████╔╝██║  ██║
        ╚═╝     ╚═╝╚═╝  ╚═╝╚═╝     ╚═╝╚═════╝ ╚═╝  ╚═╝

        mamba (0.18.2) supported by @QuantStack

        GitHub:  https://github.com/mamba-org/mamba
        Twitter: https://twitter.com/QuantStack

█████████████████████████████████████████████████████████████


     active environment : protobuf_env_mamba
    active env location : C:\Users\laboratorio\Miniconda3\envs\protobuf_env_mamba
            shell level : 1
       user config file : C:\Users\laboratorio\.condarc
 populated config files : C:\Users\laboratorio\.condarc
          conda version : 4.10.3
    conda-build version : not installed
         python version : 3.8.12.final.0
       virtual packages : __win=0=0
                          __archspec=1=x86_64
       base environment : C:\Users\laboratorio\Miniconda3  (writable)
      conda av data dir : C:\Users\laboratorio\Miniconda3\etc\conda
  conda av metadata url : None
           channel URLs : https://conda.anaconda.org/conda-forge/win-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/win-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/win-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/msys2/win-64
                          https://repo.anaconda.com/pkgs/msys2/noarch
          package cache : C:\Users\laboratorio\Miniconda3\pkgs
                          C:\Users\laboratorio\.conda\pkgs
                          C:\Users\laboratorio\AppData\Local\conda\conda\pkgs
       envs directories : C:\Users\laboratorio\Miniconda3\envs
                          C:\Users\laboratorio\.conda\envs
                          C:\Users\laboratorio\AppData\Local\conda\conda\envs
               platform : win-64
             user-agent : conda/4.10.3 requests/2.26.0 CPython/3.8.12 Windows/10 Windows/10.0.17763
          administrator : False
             netrc file : None
           offline mode : False
```
</details>

**Comment by jakirkham** [comment]
Yeah we only build the pure Python implementation on Windows. Trying to use the C++ implementation on Windows will likely run into issues for that reason.

We have tried to build the C++ implementation on Windows (and agree that would be preferable). However have run into issues in the past. For example not handling the library name correctly in the Python scripts, trouble resolving symbols, etc. Would not be surprised if this requires some patching to `setup.py`.

If this is something folks are interested in working on, we would be happy to accept a PR 🙂

**Comment by carlodri** [comment]
Thanks @jakirkham for stepping in and updating #132! I have abandoned using `protobuf` for the moment, so I don't have the time to work on it right now...

**Comment by jakirkham** [comment]
No worries. Hopefully the additional info there will be useful for whoever looks at this next

**Comment by xylar** [comment]
I believe this was fixed in #152
